src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/VirtualProject.cs

```diff
@@ -30,22 +73,29 @@ internal static (string virtualProjectXml, bool isFileBasedProgram) MakeVirtualP
         var root = tree.GetRoot();
         var isFileBasedProgram = root.GetLeadingTrivia().Any(SyntaxKind.IgnoredDirectiveTrivia) || root.ChildNodes().Any(node => node.IsKind(SyntaxKind.GlobalStatement));
 
-        var virtualProjectXml = $"""
-            <Project Sdk="Microsoft.NET.Sdk">
+        var artifactsPath = GetArtifactsPath(documentFilePath);
 
+        var virtualProjectXml = $"""
+            <Project>
                 <PropertyGroup>
                     <OutputType>Exe</OutputType>
-                    <TargetFramework>net8.0</TargetFramework>
+                    <TargetFramework>net10.0</TargetFramework>
                     <ImplicitUsings>enable</ImplicitUsings>
                     <Nullable>enable</Nullable>
                     <Features>$(Features);FileBasedProgram</Features>
                     <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
+                    <IncludeProjectNameInArtifactsPaths>false</IncludeProjectNameInArtifactsPaths>
+                    <ArtifactsPath>{SecurityElement.Escape(artifactsPath)}</ArtifactsPath>
                 </PropertyGroup>
 
+                <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
```

Comment:
This needs to be imported after ArtifactsPath is set for the Microsoft.NET.Sdk to actually respect it.

Reply:
Is this matching updates that were made to the SDK? Is anything else not lining up?

Reply:
> This needs to be imported after ArtifactsPath is set for the Microsoft.NET.Sdk to actually respect it.

I assume there's a comment in the SDK version of this to that effect?

Reply:
I went thru the run-api tests to make sure things line up quite closely, adjusted accordingly, and am manual testing now.